### PR TITLE
story2task1 - Setup the data flow for a clicked on book from (All books) to pass the data through to BookDetail

### PR DESCRIPTION
**Pull request by GDTimi**
Implementation of the clicked book through to the BookDetail component in place, open for review.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-      <BookDetail />
-    </>
-  )
+    <div>
+      <BrowserRouter>
+        <Nav />
+
+          <Routes>
+            <Route path="/" element={<HomePage />} />
+
```
Comment (Andrei-Trifu) [inline src/App.jsx]: remove these spaces and move <routes> to the same line as <nav> as Geoff said

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        .then(function (bookData) {
+            setAllBooks(bookData.data)
+            console.log(bookData.data)
+        })
+    }, [])
+    
+    return (
+        <>
+    
+            {<div className= "all-books">
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/AllBooks.jsx:25]: remove the extraspaces and the curly bracket in front of the <div>

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const[clickedBook, setClickedBook] = useState(null);
+
+    useEffect( function () {
+        fetch('https://book-swap-api.dev.io-academy.uk/api/books')
+        .then(function(res) {
+            return res.json()
+        })
+        .then(function (bookData) {
+            setAllBooks(bookData.data)
+            console.log(bookData.data)
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/AllBooks.jsx]: remove this console.log

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        })
+    }, [])
+    
+    return (
+        <>
+    
+            {<div className= "all-books">
+                {allBooks.map(
+                    bookData => 
+                    <NavLink to={"/book/" + bookData.id}>
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/AllBooks.jsx:28]: should all the details be links?

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            return res.json()
+        })
+        .then(function (bookData) {
+            setAllBooks(bookData.data)
+            console.log(bookData.data)
+        })
+    }, [])
+    
+    return (
+        <>
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/AllBooks.jsx:24]: can we remove this?

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+function BookListing({id, image, title, author, genre,}) {
+    
+    return (
+        <div className="book">
+             <img src={image} />
+            <h3>Title: {title}</h3>
+            <p>Author: {author}</p>
+            <p>Genre: {genre}</p>
+
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: remove this extra space

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    return (
+        <div className="book">
+             <img src={image} />
+            <h3>Title: {title}</h3>
+            <p>Author: {author}</p>
+            <p>Genre: {genre}</p>
+           
+        </div>
+    )
+
```
Comment (Andrei-Trifu) [inline src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: remove extra space

File: src/components/BookDetail/BookDetail.css [hunk 8]
```diff
@@ -4,20 +4,33 @@
     display:grid;
     grid-template-columns: 1fr 3fr;
     color: black;
+    /* background-color:darkslategrey; */
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.css]: remove this comment

File: src/components/BookDetail/BookDetail.css [hunk 9]
```diff
@@ -4,20 +4,33 @@
...
     border-radius: 15px;
     padding: 20px;
+    margin: 30px;
+
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.css]: remove this extra space

File: src/components/BookDetail/BookDetail.jsx [hunk 10]
```diff
@@ -23,35 +28,57 @@ function BookDetail({bookID = 4}) {
...
+                        <h1>{bookTitle}</h1>
+                        <p>{bookAuthor}</p>
+                        <p>{bookYear}</p>
+                        <p>{bookPageCount}</p>
+                        <p>{bookGenre}</p>
+                        <p>{bookBlurb}</p>
+                    </div>
+                </div>                
+            } 
+
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.jsx]: remove extra spaces

File: src/components/BookDetail/BookDetail.jsx [hunk 11]
```diff
@@ -23,35 +28,57 @@ function BookDetail({bookID = 4}) {
...
+
         </div>
-        </>
+
     )
+
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.jsx]: remove extra space

**Review by Andrei-Trifu — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (GDTimi) [reply src/App.jsx]: White space removed (and new route in place):

 <Route path="book/:id" element={<BookDetail />} />

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 2 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx:25]: Yep, done and fixed up indents.

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 4 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx:28]: Yep, we think so - the border helps denote it as a clickable object.

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 5 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx:24]: We need the fragment tags (errors without it).

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx (hunk 7 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: Yep, done.

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx (hunk 6 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: Yep, done.

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 5 above)
Comment (Andrei-Trifu) [reply src/Components/HomePage/AllBooks/AllBooks.jsx:24]: got it!

File: src/components/BookDetail/BookDetail.css (hunk 8 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.css]: Sure, done.

File: src/components/BookDetail/BookDetail.css (hunk 9 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.css]: Yep, removed.

File: src/components/BookDetail/BookDetail.jsx (hunk 10 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.jsx]: Done :)

File: src/components/BookDetail/BookDetail.jsx (hunk 11 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.jsx]: Done!

File: src/App.jsx [hunk 12]
```diff
@@ -1,14 +1,22 @@
...
-
   return (
-    <>
-      <BookDetail />
-    </>
-  )
+    <div>
+      <BrowserRouter>
+        <Nav />
+          <Routes>
```
Comment (Andrei-Trifu) [inline src/App.jsx]: still think <routes> should be at the same intend level as <nav>

**Review by Andrei-Trifu — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 12 above)
Comment (GDTimi) [reply src/App.jsx]: Yep, fixed.

**Pull request merged**